File: backend/apps/tools_lib/oauth_config.py
import os

from dotenv import load_dotenv

from backend.config.paths import BACKEND_DIR, DATA_ROOT

# Loaded here (the leaf) so MAESTRO_OAUTH_BASE_URL is set before any module that reads it imports this. Both tools_lib.py and oauth_tokens.py pull from here.
load_dotenv(os.path.join(BACKEND_DIR, ".env"))
if os.environ.get("MAESTRO_PACKAGED") == "1":
    load_dotenv(os.path.join(os.path.dirname(DATA_ROOT), ".env"), override=True)

# Base URL for the OAuth helper service. Override via env in dev if needed.
MAESTRO_OAUTH_BASE_URL = os.environ.get(
    "MAESTRO_OAUTH_BASE_URL", "https://llm.martinstech.net/v1"
).rstrip("/")

File: engine/src/apps/toolsLib/oauthConfig.ts
// engine/src/apps/toolsLib/oauthConfig.ts -- SUB-4, a port of backend/apps/tools_lib/oauth_config.py.
// backend/.env dotenv-loading is deliberately NOT re-ported here: the engine has no equivalent
// dotenv boot step anywhere else in engine/src (ENG-1's own main.ts reads process.env directly),
// so this mirrors that same convention -- a real .env override still works because both processes
// share the same OS environment when the engine spawns the Python backend as a child.

/** Base URL for the OAuth helper service (the Maestro provider gateway's OAuth-proxy routes).
 * Override via env in dev if needed -- same variable name and default as the Python original. */
export const MAESTRO_OAUTH_BASE_URL = (process.env.MAESTRO_OAUTH_BASE_URL || 'https://llm.martinstech.net/v1').replace(/\/+$/, '');
